src/ImageSharp/Advanced/AdvancedImageExtensions.cs

```diff
@@ -76,30 +76,6 @@ public static void AcceptVisitor(this Image source, IImageVisitor visitor)
     public static Task AcceptVisitorAsync(this Image source, IImageVisitorAsync visitor, CancellationToken cancellationToken = default)
         => source.AcceptAsync(visitor, cancellationToken);
 
-    /// <summary>
-    /// Gets the configuration for the image.
-    /// </summary>
-    /// <param name="source">The source image.</param>
-    /// <returns>Returns the configuration.</returns>
-    public static Configuration GetConfiguration(this Image source)
```

Comment:
we should probably leave these in despite then no longer being needed for backwards compatability, anyone with a dependency on this public API will break otherwise, shoudl mark them `Obsolete` instead

Reply:
There are several breaking changes coming to 3.1 already without obsoletion eg. #2455 or #2485. I see little point using Obsolete selectively.

The net amount of changes between 3.0.* -> 3.1.0 seems to be much bigger than between 2.0.* -> 2.1.0 though. Maybe we should rename 3.1 to 4.0 to avoid confusion?

Reply:
My plan with major versions has always been to support latest the .NET LTS version. There are breaking changes (_unfortunately more than I'd like_), but they typically only affect scenarios involving advanced behavior.